test/user.test.js

```diff
@@ -80,14 +99,21 @@ describe('User', function() {
 
     // Update the AccessToken relation to use the subclass of User
     AccessToken.belongsTo(User, {as: 'user', foreignKey: 'userId'});
+    AccessToken.belongsTo(BasicUser, {as: 'user', foreignKey: 'userId'});
+    AccessToken.belongsTo(AdminUser, {as: 'user', foreignKey: 'userId'});
```

Comment:
The setup where a single access token is shared by multiple User models triggers different code paths in many auth-related methods. We need to preserve the tests in this file to keep using a single User model.

Please add your new test(s) to [test/multiple-user-principal-types.test.js](https://github.com/strongloop/loopback/blob/c9913927e53ddf27fb5b6be211e46cfdf572380b/test/multiple-user-principal-types.test.js) instead.